spatialdata/_core/elements.py

```diff
@@ -0,0 +1,177 @@
+from abc import ABC, abstractmethod, abstractproperty
+from typing import Any, Optional, Tuple
+
+import zarr
+from anndata import AnnData
+from ome_zarr.scale import Scaler
+
+from spatialdata._core.transform import Transform
+from spatialdata._io.write import (
+    write_image,
+    write_labels,
+    write_points,
+    write_polygons,
+    write_table,
+)
+from spatialdata._types import ArrayLike
+
+__all__ = ["Image", "Labels", "Points", "Polygons"]
+
+
+class BaseElement(ABC):
+    # store the actual data (e.g., array for image, coordinates for points, etc.)
+    data: Any
+
+    # the annotation table (if present)
+    # TODO: do we want to store it here?
+    table: Optional[AnnData] = None
+
+    # store the transform objects as a dictionary with keys (source_coordinate_space, destination_coordinate_space)
+    transforms: Optional[Transform] = None
+
+    @abstractmethod
+    def to_zarr(self, group: zarr.Group, name: str, scaler: Optional[Scaler] = None) -> None:
+        """Write element to file."""
+
+    @abstractmethod
+    def transform(self, new_coordinate_space: str, inplace: bool = False) -> "BaseElement":
```

Comment:
What do you think about changing this to `transform_to()`? I think `BaseElement.transforms` is a bit to similar to `BaseElement.transform()`.

```python

# get the local to global transformation
global_transform = image.transforms["local_to_global"]

# transform to the global coordinate system
new_image = image.transform_to("global", in_place=False)
```